et/docker.ts

```diff
@@ -39,39 +39,39 @@ async function ccdLogin() {
 async function ccdInit() {
   temporaryLog('Running ./ccd init')
   const { stderr, code } = await execCommand('./ccd init', process.env.ECM_DOCKER_DIR, false)
-  if (stderr && !stderr.includes("network with name ccd-network already exists")) {
+  if (stderr && !stderr.includes('network with name ccd-network already exists')) {
     throw new Error(`./ccd init failed with exit code ${code}:  ${stderr}`)
   }
 }
 
 /**
  * Runs ccd compose up -d in the ecm-ccd-docker repo
  */
-function ccdComposeUp() {
+async function ccdComposeUp() {
   temporaryLog('Running ./compose up -d')
-  return execCommand('./ccd compose up -d', process.env.ECM_DOCKER_DIR)
+  return await execCommand('./ccd compose up -d', process.env.ECM_DOCKER_DIR)
 }
 
 /**
- * Run the init-ecm command responsible for creating roles and loading data. 
+ * Run the init-ecm command responsible for creating roles and loading data.
  * This command will automatically retry every 30 seconds (if the usual error messages occur)
  * until it is successful (can take around 5 mins depending on hardware)
  */
-function initEcm() {
-  temporaryLog("Running init-ecm.sh")
-  const promise = () => {
-    return new Promise(resolve => {
-      exec("./bin/ecm/init-ecm.sh", { cwd: process.env.ECM_DOCKER_DIR }, (error?: ExecException) => {
-        if (error?.message?.indexOf("Empty reply from server") > -1) {
-          temporaryLog(`init-ecm.sh failed with empty reply, waiting for 30s and trying again\r`)
-          return setTimeout(() => promise().then(() => resolve('')).catch(() => undefined), 1000 * 30)
+async function initEcm() {
+  temporaryLog('Running init-ecm.sh')
+  const promise = async () => {
+    return await new Promise(resolve => {
+      exec('./bin/ecm/init-ecm.sh', { cwd: process.env.ECM_DOCKER_DIR }, (error?: ExecException) => {
+        if (error?.message?.includes('Empty reply from server')) {
+          temporaryLog('init-ecm.sh failed with empty reply, waiting for 30s and trying again\r')
+          return setTimeout(() => { promise().then(() => resolve('')).catch(() => undefined) }, 1000 * 30)
         }
-        temporaryLog(`init-ecm.sh successful`)
+        temporaryLog('init-ecm.sh successful')
         resolve('')
       })
     })
   }
-  return promise()
+  return await promise()
```

Comment:
Why not make this an IIFE

Reply:
because it gets called inside itself on failure